Sources/SourceKit/SourceKitServer.swift

```diff
@@ -599,6 +599,8 @@ extension SourceKitServer {
     let index = self.workspace?.index
     let callback = callbackOnQueue(self.queue) { (result: Result<SymbolInfoRequest.Response, ResponseError>) in
       guard let symbols: [SymbolDetails] = result.success ?? nil, let symbol = symbols.first else {
+        let handled = languageService.definition(req)
```

Comment:
If symbol info failed, how is definition going to work?  I would have expected we would do this call farther down if something goes wrong with the index lookup.  For example, we already handle that in Swift with the "fallbackLocation".  Since you're doing a separate request, maybe we should refactor this code to have a fallback handler closure that either uses the fallback location in sourcekitd or calls definition in clangd.  That would also let us avoid the boolean return type on definition, you could just have the sourcekitd implementation fatalError.

Reply:
FWIW for this particular case `clangd` doesn't return any symbols, so in my mind the case is a bit different than sourcekitd's case.

A fallback handler could work, I'll try that

Reply:
Oh, what exactly is the case you're thinking of?  The case I was considering was opening a file but we have no index data.  The compiler still knows how to find the declaration in a header, so it's better than nothing even if it can't find the real definition which might be another .cpp.

Reply:
By goto-header I mean CMD+click on header `#include` or `#import` in order to open the file, e.g.

![Open header from include](https://user-images.githubusercontent.com/3173562/74786761-80962500-527b-11ea-929f-9c6cd095c3e4.png)